src/main/java/net/minecraftforge/event/entity/living/LivingKnockBackEvent.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2017.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.entity.living;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.monster.EntityMob;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.DamageSource;
+import net.minecraftforge.common.ForgeHooks;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+
+/**
+ * LivingKnockBackEvent is fired when a living entity is about to be knocked back. <br>
+ * This event is fired whenever an Entity is knocked back in
+ * {@link EntityLivingBase#attackEntityFrom(DamageSource, float)}, 
+ * {@link EntityLivingBase#blockWithShield(EntityLivingBase)}, 
+ * {@link EntityMob#attackEntityAsMob(Entity)} and 
+ * {@link EntityPlayer#attackTargetEntityWithCurrentItem(Entity)} <br>
+ * <br>
+ * This event is fired via {@link ForgeHooks#onLivingKnockBack(EntityLivingBase, Entity, float, double, double)}.<br>
+ * <br>
+ * {@link #attacker} contains the Entity that caused the knock back. <br>
+ * {@link #strength} contains the strength of the knock back. <br>
+ * {@link #ratioX} contains the x ratio of the knock back. <br>
+ * {@link #ratioZ} contains the z ratio of the knock back. <br>
+ * <br>
+ * This event is {@link Cancelable}.<br>
+ * If this event is canceled, the entity is not knocked back.<br>
+ * <br>
+ * This event does not have a result. {@link HasResult}<br>
+ *<br>
+ * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
+ **/
+@Cancelable
+public class LivingKnockBackEvent extends LivingEvent
+{
+    protected Entity attacker;
+    protected float strength;
+    protected double ratioX, ratioZ;
```

Comment:
```java
protected final Entity attacker;
protected float strength;
protected double ratioX, ratioZ;
protected final float originalStrength;
protected final double originalRatioX, originalRatioZ;

public LivingKnockBackEvent(EntityLivingBase target, Entity attacker, float strength, double ratioX, double ratioZ)
{
    super(target);
    this.attacker = attacker;
    this.originalStrength = strength;
    this.originalRatioX = ratioX;
    this.originalRatioZ = ratioZ;
    this.strength = strength;
    this.ratioX = ratioX;
    this.ratioZ = ratioZ;
}
```
In case of multiple mods modifying the same event.

Reply:
This is a good suggestion from liach, please add the "original" values as final values, so mods using the events can have more information about modifications to the values.

Reply:
Fixed now.